### PR TITLE
[0.2] Settings - Manage Product Options

### PR DESCRIPTION
**Pull request by adam-code-labx**
See #406 
Duplicated Product Feature section to manage product options.

Screenshot 2022-07-26 at 00 32 54

Validation and tests + refactor still to do so opening up this PR in draft.

### CODE REVIEW

**Comment by glennjacobs** [comment]
We'll have to manually test this to ensure it's all good, but I do welcome the idea of editing product options globally in the settings area.

**Comment by adam-code-labx** [comment]
Sure there is a still a little more work on this one as currently copied over from manage features should be completed by tomorrow. Would be nice to see both the product options and features at least get merged to the new develop branch.

**Comment by adam-code-labx** [comment]
@glennjacobs Ok this one is ready for review.
I removed the manage features and created a few tests so should be good for the next beta.
Unless there is anything else you need me to add for this one?

**Review by markmead — CHANGES_REQUESTED** [review]
Hey, @adam-code-labx I'm not too clued up on this PR so my testing comes from being an actual user, here are a few things I spotted.

- How do you know what options have been added to a variant? 
There doesn't seem to be anything in the UI to tell me.
- How do I add options to a variant that already exists?
- When creating a new option value, I am shown this error.
`Call to undefined method GetCandy\Models\ProductOptionValue::productOption()`

Some of the UI looked off as well.

Extra border under the "Delete option" button.

Screenshot 2022-09-02 at 09 48 43

Extra border under the "Delete option value" button.

Screenshot 2022-09-02 at 09 48 49

When pressing "Enter" after filling out the first input it should submit the form, currently, it shows the next translation.

Screenshot 2022-09-02 at 09 49 59

Should this be empty? And there seems to be a missing translation.

Screenshot 2022-09-02 at 09 50 17

---

I seem to remember seeing a way to associate options with different models, has that been removed?

**Comment by adam-code-labx** [comment]
Thanks @markmead I will pull this branch down over the weekend and update those issues you mentioned.

Was thinking about perhaps extracting this to an abstract ListBuilder similar to #481 then setup each list as a component where ever we need it, for example Product features / Product options / Collections as there is currently far too much responsibility handled directly in the blade templates. Would be good to extract the actions as well.

Regarding this question
"How do I add options to a variant that already exists?"

Not quite sure what you mean on this one, as this is a global way to manage product options, at this level I don't think we need to be concerned about the variants.

**Comment by markmead** [comment]
Sorry @adam-code-labx, that question is irrelevant. I was getting confused between options and attributes, my bad!

**Comment by glennjacobs** [comment]
Hi @adam-code-labx any update on this from @markmead 's review above?

**Comment by alecritson** [comment]
@markmead @adam-code-labx 

I've done some tweaks to this this morning to try push it over the line as I think it's good enough to get in and be a useful addition to the hub, we can always improve on it down the line.

Changes:

- Add `handle` input to allow users to specify this manually, rather than auto-generating it.
- Addressed style updates as Mark mentioned
- Fixed some translations and the incorrect relationship reference.

**Comment by glennjacobs** [comment]
Tried to add two options both with the same name

image

**Comment by glennjacobs** [comment]
Clicking "Add option value" does nothing

image

**Comment by glennjacobs** [comment]
Capitalisation isn't consistent

image

**Comment by glennjacobs** [comment]
Can we show a count of the values for each option?

image

File: packages/admin/resources/views/livewire/components/settings/product/options/index.blade.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            <div class="w-full text-center text-gray-500">
+                {{ __('adminhub::components.option.no_groups') }}
+            </div>
+        @endforelse
+    </div>
+
+    <x-hub::modal.dialog wire:model="showOptionCreate" form="createOption">
+        <x-slot name="title">{{ __('adminhub::components.option.create_title') }}</x-slot>
+        <x-slot name="content">
+            <x-hub::input.group label="Name" for="optionName" :error="$errors->first('newProductOption.name.'.$this->defaultLanguage->code)">
```
Comment (glennjacobs) [inline packages/admin/resources/views/livewire/components/settings/product/options/index.blade.php:76]: Should "Name" be translated?

File: packages/admin/resources/views/livewire/components/settings/product/options/index.blade.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                <div class="flex space-x-4">
+                  <div class="relative w-full">
+                    <x-hub::translatable :expanded="false">
+                      <x-hub::input.text
+                        wire:model="newProductOption.name.{{ $this->defaultLanguage->code }}"
+                        :error="$errors->first('name.'.$this->defaultLanguage->code)"
+                        placeholder="Size"
```
Comment (glennjacobs) [inline packages/admin/resources/views/livewire/components/settings/product/options/index.blade.php:83]: Placeholder not translated

File: packages/admin/resources/views/livewire/components/settings/product/options/index.blade.php [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                            {{ $message }}
+                        </x-hub::alert>
+                        </div>
+                    @enderror
+                  </div>
+                </div>
+            </x-hub::input.group>
+        </x-slot>
+        <x-slot name="footer">
+            <x-hub::button>Create option</x-hub::button>
```
Comment (glennjacobs) [inline packages/admin/resources/views/livewire/components/settings/product/options/index.blade.php:107]: Translation required

**Review by glennjacobs — APPROVED** [review]

**Pull request merged**